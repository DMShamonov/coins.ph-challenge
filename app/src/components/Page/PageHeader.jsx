// @flow

import * as React from 'react';
import classnames from 'classnames';

import type { PageHeaderPropsType } from './types.js.flow';

export default function PageHeader({
  className,
  children,
}: PageHeaderPropsType): React.Node {
  return (
    <header className={classnames('page__header', className)}>
      {children}
    </header>
  );
}
