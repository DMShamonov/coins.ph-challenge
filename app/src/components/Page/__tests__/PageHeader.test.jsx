import React from 'react';
import { shallow } from 'enzyme';

import PageHeader from '../PageHeader';

describe('`PageHeader` component should render correctly', () => {
  test('with default props', () => {
    expect(shallow(<PageHeader>__CONTENT__</PageHeader>)).toMatchSnapshot();
  });

  test('with `className` prop', () => {
    expect(shallow(<PageHeader className="__CLASSNAME__">__CONTENT__</PageHeader>)).toMatchSnapshot();
  });
});
