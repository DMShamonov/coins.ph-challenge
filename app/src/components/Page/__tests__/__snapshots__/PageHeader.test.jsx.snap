// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`\`PageHeader\` component should render correctly with \`className\` prop 1`] = `
<header
  className="page__header __CLASSNAME__"
>
  __CONTENT__
</header>
`;

exports[`\`PageHeader\` component should render correctly with default props 1`] = `
<header
  className="page__header"
>
  __CONTENT__
</header>
`;
